### ISSUE TITLE
DEPTH_TEXTURE incorrect data

### ISSUE THREAD

**Issue by kaadmy** [godotengine/godot#25535]
**Godot version:**
[Godot 3.1-beta3](https://downloads.tuxfamily.org/godotengine/3.1/beta3/Godot_v3.1-beta3_x11.64.zip)
Godot git 463123a661151f119132f2ee9af78925a58a068a

**OS/device including version:**
```
OS: Arch Linux x86_64
Kernel: 4.19.4-arch1-1-ARCH
GPU: NVIDIA GeForce 940MX
GPU: Intel HD Graphics 620
```

**Issue description:**
In a `ShaderMaterial`, `DEPTH_TEXTURE` seems to have incorrect data (usually 0.0 or 1.0, but sometimes glitches out with a couple random pixels with arbitrary values) with the GLES3 backend. With a similar project running Godot 3.0.6-stable (not the same project, since 3.1 didn't seem backwards compatible), the same shader code works properly (you can see geometry depth behind the plane if you put the camera close enough.) `SCREEN_TEXTURE` still works properly in 3.1-beta3 as well as 3.0.6-stable.

**Minimal reproduction project:**
[depth_texture_test_case.zip](https://github.com/godotengine/godot/files/2820508/depth_texture_test_case.zip)

**Comment by clayjohn** [comment]
I can confirm this bug. I tested on a few versions (3.06, a1, b1, b2) and the bug was only present in beta3.

**Comment by kaadmy** [comment]
677d4400de9ae73713be9f33600f59a10a079c19 is the faulty commit. Adding an `OmniLight` with contact shadows enabled does indeed fix the issue.

This can likely be fixed by using a check similar to `SCREEN_TEXTURE` for `DEPTH_TEXTURE`, ie. writing to it when a shader uses it.

**Comment by kaadmy** [comment]
This issue should be fixed with #25557

**Comment by akien-mga** [comment]
Reopening as #25557 was reverted in 0eb61c3106 as it triggered #26258 (and @reduz says it's not the correct fix for this issue).

**Comment by kaadmy** [comment]
Out of curiosity, what would a more correct fix for #25557 be?

**Comment by clayjohn** [comment]
@akien-mga I think this should be part of the 3.1 milestone. Or at least part of an early round of bug fixes. Reading from depth is important for a lot of SpatialMaterial effects (depth fade, proximity fade, etc.).

**Comment by Nophlock** [comment]
I don't know if this is related to this problem, but I wanted to play around a bit with the Depth-buffer in an Shader and found this in the Godot-Asset-Library: https://godotengine.org/asset-library/asset/76
After downloading it, I stumbled upon the issue, that the intersection parts are not highlighted in Godot 3.1-Stable (this actually use the Depth-buffer to achieve the white borders) but is correctly working with Godot 3.0.6. So I guess this a bug and hopefully related to this issue here. I've uploaded two screenshots to make it more clear what I mean. 

Godot 3.0.6-Stable
![image](https://user-images.githubusercontent.com/8605748/54439946-148d7180-473a-11e9-8075-8e6e5024ced3.png)

Godot 3.1.0-Stable
![image](https://user-images.githubusercontent.com/8605748/54440015-2f5fe600-473a-11e9-9c84-3b9be31f04fc.png)
